### PR TITLE
fix(github): adapt bp v100 to v200

### PR DESCRIPTION
**Pull request by warren830**
### Summary
In this pr, we mainly work on let bp v100 and advanced mode can work properly in v0.15

In addition, we canceled setting a repo to githubTaskData when preparing, instead, we set githubId for githubOptions, consequently, we modified the relevant value in all tasks and e2e

### Does this close any open issues?
Relate to #3468 

### Screenshots
Our input for bp v100:
```json
{
  "name": "test blueprint normal",
  "mode": "NORMAL",
  "settings": {
    "version": "1.0.0",
    "connections": [
      {
        "plugin": "github",
        "connectionId": 1,
        "scope": [
          {
            "options": {
              "owner": "warren830",
              "repo": "gorm"
            },
            "transformation": {
              "prType": "hey,man,wasup",
              "refdiff": {
                "tagsPattern": "pattern",
                "tagsLimit": 10,
                "tagsOrder": "reverse semver"
              },
              "productionPattern": "xxxx"
            }
          }
        ]
      }
    ]
  },
  "isManual": true
}
```
we can succeed to create bp
and blow is the screenshot for the result of trigger:
image
image
 
advanced mode:
image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: models/domainlayer/devops/cicd_pipeline_commmit.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ type CiCDPipelineCommit struct {
 	CommitSha  string `gorm:"primaryKey;type:varchar(255)"`
 	Branch     string `gorm:"type:varchar(255)"`
 	RepoId     string `gorm:"index;type:varchar(255)"`
-	RepoUrl    string
+	Repo       string
```
Comment (warren830) [inline models/domainlayer/devops/cicd_pipeline_commmit.go:28]: This should be changed long ago...

File: plugins/bitbucket/tasks/pipeline_convertor.go [hunk 2]
```diff
@@ -77,7 +77,7 @@ func ConvertPipelines(taskCtx core.SubTaskContext) errors.Error {
 					Generate(bitbucketPipeline.ConnectionId, bitbucketPipeline.RepoId),
 				CommitSha: bitbucketPipeline.CommitSha,
 				Branch:    bitbucketPipeline.RefName,
-				RepoUrl:   bitbucketPipeline.WebUrl,
+				Repo:      bitbucketPipeline.WebUrl,
```
Comment (warren830) [inline plugins/bitbucket/tasks/pipeline_convertor.go:80]: This should be changed long ago...

File: plugins/dora/e2e/raw_tables/cicd_pipeline_commits.csv [hunk 3]
```diff
@@ -1,4 +1,4 @@
-pipeline_id,commit_sha,branch,repo_id,repo_url
+pipeline_id,commit_sha,branch,repo_id,repo
```
Comment (warren830) [inline plugins/dora/e2e/raw_tables/cicd_pipeline_commits.csv:1]: This should be changed long ago...

File: plugins/github/api/blueprint.go [hunk 4]
```diff
@@ -65,11 +61,11 @@ func MakePipelinePlan(subtaskMetas []core.SubTaskMeta, connectionId uint64, scop
 	return plan, nil
 }
 
-func makePipelinePlan(subtaskMetas []core.SubTaskMeta, scope []*core.BlueprintScopeV100, apiClient helper.ApiClientGetter, connection *models.GithubConnection) (core.PipelinePlan, errors.Error) {
+func makePipelinePlan(subtaskMetas []core.SubTaskMeta, scopeV100s []*core.BlueprintScopeV100, apiClient helper.ApiClientGetter, connection *models.GithubConnection) (core.PipelinePlan, errors.Error) {
```
Comment (warren830) [inline plugins/github/api/blueprint.go:64]: better name

File: plugins/github/api/blueprint.go [hunk 5]
```diff
@@ -43,14 +43,10 @@ func MakePipelinePlan(subtaskMetas []core.SubTaskMeta, connectionId uint64, scop
...
 	}
-	token := strings.Split(connection.Token, ",")[0]
-
 	apiClient, err := helper.NewApiClient(
 		context.TODO(),
 		connection.Endpoint,
-		map[string]string{
-			"Authorization": fmt.Sprintf("Bearer %s", token),
-		},
+		nil,
```
Comment (warren830) [inline plugins/github/api/blueprint.go:49]: we don't need token here

File: plugins/github/api/blueprint.go [hunk 6]
```diff
@@ -118,7 +114,7 @@ func makePipelinePlan(subtaskMetas []core.SubTaskMeta, scope []*core.BlueprintSc
 			return nil, err
 		}
 		// collect git data by gitextractor if CODE was requested
-		repo, err = memorizedGetApiRepo(repo, op, apiClient)
+		repo, err = MemorizedGetApiRepo(repo, op, apiClient)
```
Comment (warren830) [inline plugins/github/api/blueprint.go:117]: this MemorizedGetApiRepo will also be used in impl.PrepareTaskData

File: plugins/github/api/blueprint.go [hunk 7]
```diff
@@ -139,7 +135,7 @@ func makePipelinePlan(subtaskMetas []core.SubTaskMeta, scope []*core.BlueprintSc
 			if j == len(plan) {
 				plan = append(plan, nil)
 			}
-			repo, err = memorizedGetApiRepo(repo, op, apiClient)
+			repo, err = MemorizedGetApiRepo(repo, op, apiClient)
```
Comment (warren830) [inline plugins/github/api/blueprint.go:138]: this MemorizedGetApiRepo will also be used in impl.PrepareTaskData

File: plugins/github/api/blueprint.go [hunk 8]
```diff
@@ -246,7 +242,7 @@ func getApiRepo(op *tasks.GithubOptions, apiClient helper.ApiClientGetter) (*tas
 	return repoRes, nil
 }
 
-func memorizedGetApiRepo(repo *tasks.GithubApiRepo, op *tasks.GithubOptions, apiClient helper.ApiClientGetter) (*tasks.GithubApiRepo, errors.Error) {
+func MemorizedGetApiRepo(repo *tasks.GithubApiRepo, op *tasks.GithubOptions, apiClient helper.ApiClientGetter) (*tasks.GithubApiRepo, errors.Error) {
```
Comment (warren830) [inline plugins/github/api/blueprint.go:245]: this MemorizedGetApiRepo will also be used in impl.PrepareTaskData

File: plugins/github/api/blueprint_V200_test.go [hunk 9]
```diff
@@ -149,7 +148,6 @@ func NewMockBasicRes() *mocks.BasicRes {
 		GithubId:             12345,
 		Name:                 "test/testRepo",
 		TransformationRuleId: 1,
-		CreatedDate:          time.Time{},
```
Comment (warren830) [inline plugins/github/api/blueprint_V200_test.go:152]: Already changed the type of CreatedDate to pointer

File: plugins/github/api/scope.go [hunk 10]
```diff
@@ -70,7 +70,8 @@ func PutScope(input *core.ApiResourceInput) (*core.ApiResourceOutput, errors.Err
 			keeper[repo.GithubId] = struct{}{}
 		}
 		repo.ConnectionId = connectionId
-		repo.CreatedDate = now
+		// Fixme: why do we set this to now?
+		repo.CreatedDate = &now
```
Comment (warren830) [inline plugins/github/api/scope.go:74]: Actually, I have no idea why do we set the value here

File: plugins/github/impl/impl.go [hunk 11]
```diff
@@ -169,14 +174,9 @@ func (plugin Github) PrepareTaskData(taskCtx core.TaskContext, options map[strin
...
 		}
 	}
-	apiClient, err := tasks.CreateApiClient(taskCtx, connection)
-	if err != nil {
-		return nil, errors.Default.Wrap(err, "unable to get github API client instance")
-	}
 	taskData := &tasks.GithubTaskData{
 		Options:   op,
 		ApiClient: apiClient,
-		Repo:      githubRepo,
```
Comment (warren830) [inline plugins/github/impl/impl.go:179]: We don't need set repo here, we only need to set githubId to options

**Review by warren830 — COMMENTED** [review]
Hope these comments can help review

**Review by likyh — APPROVED** [review]

**Review by likyh — APPROVED** [review]

**Pull request merged**